src/pages/layout.js

```diff
@@ -96,48 +93,42 @@ class Layout extends React.Component {
     componentDidMount() {
         api.getNamespaces()
             .then(res => {
-                const nameSpace = res.filter(name => name === NAMESPACE)
-                nameSpace.length === 0
-                    ? this.setState({
-                          openFirstLaunch: true,
-                      })
-                    : this.setState({
-                          openFirstLaunch: false,
-                          isSaved: true,
-                      })
+                if (res.includes(NAMESPACE)) {
+                    this.setState({
+                        openFirstLaunch: false,
+                        isSaved: true,
+                    })
+                }
             })
             .catch(e => {
                 console.error(e)
             })
     }
 
     render() {
-        if (this.state.openFirstLaunch === true) {
-            return (
-                <DialogFirstLaunch
-                    openDialog={this.state.openFirstLaunch}
-                    onClose={this.handleClose}
-                    handleSave={this.handleSave}
-                />
-            )
-        }
-
         return (
             <HashRouter>
                 <TwoPanel mainStyle={styles.twoPanelMain}>
                     <SideBar />
                     <MainContent>
                         <Paper className={layoutStyles.paper__layout}>
                             <Switch>
-                                <Route
-                                    path="/"
-                                    exact
-                                    render={() => (
+                                <Route path="/" exact>
+                                    {this.state.openFirstLaunch === true ? (
+                                        <D2Shim>
```

Comment:
Is it possible to have the <D2Shim> component wrap the ternary statement all together?

```
<D2Shim>
{this.state.openFirstLaunch === true ? (...) : (...)}</D2Shim>

Reply:
Just checked this option and it's working